modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/APMAgentSettings.java

```diff
@@ -132,7 +248,7 @@ public void setAgentSetting(String key, String value) {
             final String key = parts[parts.length - 1];
             final String defaultValue = APM_AGENT_DEFAULT_SETTINGS.getOrDefault(key, "");
             return new Setting<>(qualifiedKey, defaultValue, (value) -> {
-                if (PROHIBITED_AGENT_KEYS.contains(key)) {
+                if (qualifiedKey.equals("_na_") == false && PERMITTED_AGENT_KEYS.contains(key) == false) {
```

Comment:
I think the `_na_` had to do with `Setting.affixKeySetting`'s `delegateFactory` which is used via `Setting.prefixKeySetting` by `APM_AGENT_SETTINGS`.  Are you sure it's ok to remove this check?

Reply:
Moving from deny to allow list, I had to explicitly allow `_na_`. As you mentioned the `delegateFactory` is always invoked once with `_na_`, which obviously shouldn't fail.